src/main.js

```diff
@@ -19,22 +20,22 @@ const listFunctionsFiles = async function (
   srcFolder,
   { jsBundler = JS_BUNDLER_ZISI, jsExternalModules, jsIgnoredModules } = {},
 ) {
-  const [functionInfos, pluginsModulesPath] = await Promise.all([
-    getFunctionInfos(srcFolder),
-    getPluginsModulesPath(srcFolder),
-  ])
+  const paths = await listFunctionsDirectory(srcFolder)
+  const functions = await runtimes.getFunctionsFromPaths(paths)
+  const pluginsModulesPath = await getPluginsModulesPath(srcFolder)
   const listedFunctionsFiles = await Promise.all(
-    functionInfos.map((info) =>
+    [...functions.values()].map((info) =>
       getListedFunctionFiles(info, { jsBundler, jsExternalModules, jsIgnoredModules, pluginsModulesPath }),
     ),
   )
+
   // TODO: switch to Array.flat() once we drop support for Node.js < 11.0.0
   // eslint-disable-next-line unicorn/prefer-spread
   return [].concat(...listedFunctionsFiles)
 }
 
 const getListedFunction = function ({ runtime, name, mainFile, extension }) {
-  return { name, mainFile, runtime, extension }
+  return { name, mainFile, runtime: runtime.name, extension }
```

Comment:
The `runtime` object that is passed around used to be a String containing the name of the runtime. It is now an object (or an "instance" of a runtime). We convert it to the string containing the name only at the termination points, like this one.